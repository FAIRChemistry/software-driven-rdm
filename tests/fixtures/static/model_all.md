# Test

This model tests all methods and types that are present within the sdRDM library.

## Objects

### Root

- str_value
  - Type: string
- float_value
  - Type: float
- int_value
  - Type: integer
- bool_value
  - Type: boolean
- date_value
  - Type: date
- datetime_value
  - Type: datetime
- posfloat_value
  - Type: posfloat
- posint_value
  - Type: PositiveInt
- http_url_value
  - Type: AnyHttpUrl
- email_value
  - Type: EmailStr
- bytes_value
  - Type: bytes
- multiple_primitives
  - Type: float
  - Multiple: True
- enum_value
  - Type: SomeEnum
- nested_single_obj
  - Type: Nested
- nested_multiple_obj
  - Type: Nested
  - Multiple: True
- referenced_value
  - Type: @Nested.id

### Nested

- str_value
  - Type: string
- float_value
  - Type: float
- int_value
  - Type: integer

## Enumerations

### SomeEnum

```python
VALUE1 = "value1"
VALUE2 = "value2"
```
