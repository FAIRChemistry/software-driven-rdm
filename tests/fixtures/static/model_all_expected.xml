<?xml version='1.0' encoding='UTF-8'?>
<Root id="id">
  <string>string</string>
  <float>1.5</float>
  <integer>1</integer>
  <boolean>true</boolean>
  <date>2023-12-17</date>
  <datetime>2023-12-17T15:55:44.245441</datetime>
  <posfloat>1.5</posfloat>
  <PositiveInt>2</PositiveInt>
  <AnyHttpUrl>https://www.google.com/</AnyHttpUrl>
  <EmailStr>max.mustermann@muster.de</EmailStr>
  <bytes>this_is_bytes</bytes>
  <multiple_primitives>
    <float>1.5</float>
    <float>1.7</float>
    <float>1.9</float>
  </multiple_primitives>
  <SomeEnum>value1</SomeEnum>
  <Nested id="id">
    <string>string</string>
    <float>1.5</float>
    <integer>1</integer>
  </Nested>
  <nested_multiple_obj>
    <Nested id="id">
      <string>string</string>
      <float>1.5</float>
      <integer>1</integer>
    </Nested>
  </nested_multiple_obj>
</Root>
